reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,829 @@
+                raise RuntimeError(msg)
+        else:
+            time_index = solar_time_index
+
+        return time_index, solar_time_slice, wind_time_slice
+
+    @staticmethod
+    def _pearson_correlation(solar_ts, wind_ts):
+        """
+        Compute the Pearson correlation coefficient between the given solar
+        and wind timeseries data. Return just the correlation coefficient.
+
+        Parameters
+        ----------
+        solar_ts : ndarray
+            Solar time-series vector for a single site
+        wind_ts : ndarray
+            Wind time-series vector for a single site
+
+        Returns
+        -------
+        float
+            Pearson correlation coefficient
+        """
+        return pearsonr(solar_ts, wind_ts)[0]
+
+    @staticmethod
+    def _spearman_correlation(solar_ts, wind_ts):
+        """
+        Compute the Spearman correlation coefficient between the given solar
+        and wind timeseries data. Return just the correlation coefficient.
+
+        Parameters
+        ----------
+        solar_ts : ndarray
+            Solar time-series vector for a single site
+        wind_ts : ndarray
+            Wind time-series vector for a single site
+
+        Returns
+        -------
+        float
+            Spearman correlation coefficient
+        """
+        return spearmanr(solar_ts, wind_ts)[0]
+
+    @staticmethod
+    def _kendall_tau(solar_ts, wind_ts):
+        """
+        Compute Kendall's tau correlation between the given solar
+        and wind timeseries data. Return just the correlation coefficient.
+
+        Parameters
+        ----------
+        solar_ts : ndarray
+            Solar time-series vector for a single site
+        wind_ts : ndarray
+            Wind time-series vector for a single site
+
+        Returns
+        -------
+        float
+            Kendall's tau
+        """
+        return kendalltau(solar_ts, wind_ts)[0]
+
+    @staticmethod
+    def _groupby_data(data, diurnal=False, month=False):
+        """
+        Groupby data by month and/or hour
+
+        Parameters
+        ----------
+        data : pandas.DataFrame
+            DataFrame of data where index is time_index, columns are sites
+        diurnal : bool, optional
+            Extract diurnal stats, by default False
+        month : bool, optional
+            Extract monthly stats, by default False
+
+        Returns
+        -------
+        data : pandas.Groupby
+            Input DataFrame grouped by month and or hour if requested
+        """
+        groupby = [data.index.year]
+        if month:
+            groupby.append(data.index.month)
+
+        if diurnal:
+            groupby.append(data.index.hour)
+
+        data = data.groupby(groupby)
+
+        return data
+
+    @staticmethod
+    def _format_grp_names(index):
+        """
+        Format groupby index values
+
+        Parameters
+        ----------
+        index : list
+            Group by index values
```

Comment:
This whole thing is confusing to me. What index is this?

Reply:
I'll re-name, index is the groupby names. I was actually proud of this code, but yes it is confusing as F...
The goal is to properly convert the group names into year-month-day-hour (at most) that are all unique